use partitioner instance directly in to_backend

cccclai opened the pull request:
Summary: to_backend either takes partitioner or a dict of partitioner `key: method_name, value: partitioner`. We shouldn't do key as the backend name and value as the partitioner.

Differential Revision: D55078939

mergennachin reviewed (approved):
See inline in phabricator

cccclai committed:
use partitioner instance directly in to_backend (#2513)

Summary:

to_backend either takes partitioner or a dict of partitioner `key: method_name, value: partitioner`. We shouldn't do key as the backend name and value as the partitioner.

Reviewed By: mergennachin

Differential Revision: D55078939
